docs/tutorial/2_MarketMaker.md

```diff
@@ -0,0 +1,230 @@
+
+
+Just like I did in part one of the tutorial, I need to approve `MarketMaker` to use my CLN. I call `approve` function of [ColuLocalNetwork](../reference/ColuLocalNetwork.md) contract, fill in the amount I wish to exchange and specify `MarketMaker` address as the spender. If you don't sure why we're doing this, please read "A Few Words About ERC20" section of the first part.
+
+Investigating [transaction's](https://ropsten.etherscan.io/tx/0x4d59b4e0dfe3e853e94e0515bda6a0cac921b5db54ad52b2edf950d7c1c574d4) arguments you can see that the first one is my `MarketMaker` address - `000000000000000000000000b3f9a85d00fcb75be507da5efc0b91ed221e9bb9` and the second is the amount in hex (0xde0b6b3a7640000 = 1e18).
+
+
+The function that exchanges CLN/CC is called, surprisingly, `change`. There's multiple implementations with different arguments, I choose one that takes 3 arguments:
+-  `_fromToken` -  [ColuLocalNetwork](../reference/ColuLocalNetwork.md) address.
+-  `_inAmount` - the amount of tokens I want to exchange.
+-  `_toToken` - the address of my `CommunityCurrency`
+
+![mew_change](../assets/mew_change.png)
+
+Let's sign and send the transaction. Oh, look at [that](https://ropsten.etherscan.io/tx/0x9e3ef01e47e4a1d6af4d1fbcbcca6f0b7ed287476bddd7a26c174e97ae68788d), I did receive ~0.57 CC in exchange of 1 CLN.
+
+Let's check again the CLN/CC exchange rate, I call `getCurrentPrice` again and get a slightly different price! It returns now 569388278636115886, the delta is 569672901914775677 - 569388278636115886 = 284623278659791. We need to divide it by 1e18, so it gives a change of 0.00028. This means that now we receive less CC for CLN. It's small, but we see that our CC it's getting more expensive. That's the Mechanical Market Maker in action
+
+The most observant of you might noticed that in the last transaction I exchanged 1 CLN for 0.56953055471532534 CC. Let's multiply it by 1e18 to align with other values. 0.56953055471532534 * 1e18 = 569530554715325340, I call this `actualRate`. Then:
+
+> 569672901914775677 > 569530554715325340 > 569388278636115886
+
+Which means:
+
+> getCurrentPrice_before > actualRate > getCurrentPrice_after
+
+I see that actually I exchanged for a slightly different rate than `getCurrentPrice`. Is someone fools me around :monkey_face:?
+
+The exchange rate of CLN/CC is a function with multiple variables, we get to this later. What's important to grasp is that it defined continuously for every wei (smallest indivisible unit) in the domain. So the exchange rate changes with every inserted wei of CLN. When I inserted 1 CLN It was actually 1e18 wei, and every wei got exchanged for an individual price. If it reminds you of something you're right - it's good old calculus. To calculate the `actualRate`, I need to calculate a definite integral of the `getCurrentPrice` function. I stop myself here and refer you to [whitepaper's appendix](https://cln.network/pdf/cln_whitepaper.pdf) for a complete explanation.
```

Comment:
* we get to this later --> we'll get to this later
* it defined continuously --> it's defined continuously